### PR TITLE
migrator/indexer: change how records are indexed

### PR DESCRIPTION
**Pull request by pazembrz**
* Delete records from ES if it is marked as deleted
* Process references of records if migrator is not processing all records from mirror table
* Fix jobs and Authors to index properly

### CODE REVIEW

File: inspirehep/migrator/tasks.py [hunk 1]
```diff
@@ -243,38 +259,61 @@ def create_records_from_mirror_recids(recids):
...
-            with db.session.begin_nested():
-                record = InspireRecord.get_record(uuid)
-                if record and not record.get("deleted", False):
-                    index_queue.append(
-                        InspireRecordIndexer()._process_bulk_record_for_index(record)
+def process_references_in_records(uuids):
+    references_to_reindex = []
+    try:
+        for uuid in uuids:
+            if not all:
```
Comment (drjova) [inline inspirehep/migrator/tasks.py]: the ```all`` is undefined

File: inspirehep/migrator/tasks.py [hunk 2]
```diff
@@ -114,32 +113,49 @@ def migrate_from_mirror(also_migrate=None, disable_orcid_push=True):
             migrated.
         disable_orcid_push (bool): flag indicating whether the orcid_push
             should be disabled (if True) or executed at the end of migrations (if False).
+        disable_references_processing (bool): Flags which indicates is whole db is remigrated or not
```
Comment (ammirate) [inline inspirehep/migrator/tasks.py]: this flag in not in the args and is missing `recids`

File: inspirehep/migrator/tasks.py [hunk 3]
```diff
@@ -243,38 +267,60 @@ def create_records_from_mirror_recids(recids):
 
 @shared_task(ignore_results=False, queue="migrator", acks_late=True)
 def recalculate_citations(uuids):
-    processed_uuids = []
+    """Task which updates records_citations table with references of this record
+
+    Args:
+        recids: records uuids which references should be added to table
```
Comment (ammirate) [inline inspirehep/migrator/tasks.py]: `uuids` instead of recids

File: inspirehep/migrator/tasks.py [hunk 4]
```diff
@@ -105,41 +104,58 @@ def migrate_record_from_legacy(recid):
...
+def migrate_from_mirror(also_migrate=None, disable_orcid_push=True, recids=None):
     """Migrate legacy records from the local mirror.
     By default, only the records that have not been migrated yet are migrated.
+
     Args:
         also_migrate(Optional[string]): if set to ``'broken'``, also broken
             records will be migrated. If set to ``'all'``, all records will be
             migrated.
         disable_orcid_push (bool): flag indicating whether the orcid_push
             should be disabled (if True) or executed at the end of migrations (if False).
```
Comment (michamos) [inline inspirehep/migrator/tasks.py]: missing `recids` explanation

Comment (pazembrz) [reply inspirehep/migrator/tasks.py]: discussed IRL

**Comment by pazembrz** [comment]
Micha wants to review it on Monday. Do not merge it yet.

File: inspirehep/records/indexer/utils.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    if record_version and record.model.version_id < record_version:
+        logger.info(
+            f"Cannot pull record {uuid} in version {record_version}."
+            f"Current version: {record.model.version_id}."
+        )
+        raise StaleDataError()
+    return record
+
+
+def get_modified_references_uuids(record):
```
Comment (michamos) [inline inspirehep/records/indexer/utils.py]: is this even needed? AFAICT, this was moved from  `inspirehep.records.indexer.tasks.process_references_for_record`, but no calling code was modified anywhere.

Comment (drjova) [reply inspirehep/records/indexer/utils.py]: AFAIS is been used in couple of places

**Comment by drjova** [comment]
@michamos updated

File: tests/integration/migrator/test_migrator_tasks.py [hunk 6]
```diff
@@ -87,7 +91,7 @@ def test_migrate_and_insert_record_dojson_error(mock_logger, base_app, db):
     assert prod_record.marcxml == raw_record
 
     assert not mock_logger.error.called
-    mock_logger.exception.assert_called_once_with("Migrator DoJSON Error")
+    mock_logger.exception.assert_called_once()
```
Comment (michamos) [inline tests/integration/migrator/test_migrator_tasks.py]: you can put this back (with an extra `.` now)

File: tests/integration/migrator/test_migrator_tasks.py [hunk 7]
```diff
@@ -175,7 +179,7 @@ def test_migrate_and_insert_record_other_exception(mock_logger, base_app, db):
     assert prod_record.marcxml == raw_record
 
     assert not mock_logger.error.called
-    mock_logger.exception.assert_called_once_with("Migrator Record Insert Error")
+    mock_logger.exception.assert_called_once()
```
Comment (michamos) [inline tests/integration/migrator/test_migrator_tasks.py]: same

**Pull request merged**